src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,225 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            first = new Node<>(null, value, null);
+            last = first;
+        } else {
+            last.next = new Node<>(last, value, null);
+            last = last.next;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        indexAddValidatorForAddElement(index);
+
+        if (index == 0 && size == 0) {
+            addElementToEmptyList(value, index);
+            return;
+        }
+
+        if (index == 0) {
+            addElementToStart(value);
+            return;
+        }
+
+        if (index == size) {
+            add(value);
+            return;
+        }
+        addElementToMiddle(value, index);
```

Comment:
```suggestion
        Node<T> target = getNode(index);
        addBefore(value, target);
```
You need to create `getNode(index);` method and it will be used in many places.